core_lib/src/interface/scribblearea.cpp

```diff
@@ -383,12 +383,15 @@ void ScribbleArea::onObjectLoaded()
 
 bool ScribbleArea::event(QEvent *event)
 {
+    bool processed = false;
     if (event->type() == QEvent::WindowDeactivate)
     {
         editor()->tools()->clearTemporaryTool();
+        processed = true;
     }
 
-    return currentTool()->event(event) || QWidget::event(event);
+    processed = currentTool()->event(event) || processed;
+    return QWidget::event(event) || processed;
```

Comment:
Same here, shouldn't we utilize the event->accept/ignore state instead of returning our own processed value, I'm pretty sure it's important that no matter what we call QWidget::event, so that events are always processed by the parent?

Reply:
> QWidget::event is currently called no matter what.

Ah, my bad. I misunderstood the return part. 

As for accepted() being default `true`, the documentation also says not to rely on this.
"By default, isAccepted() is set to true, but don't rely on this as subclasses may choose to clear it in their constructor"
As such it might be a good idea to call it explicitly still?

> You see this is exactly the problem which tripped me up as well, the return value and the accept/ignore state serve entirely different purposes:

Ah... Thanks for clearing that up, that confused me too! I had to look at the QWidget source code to understand it completely.